roc.go

```diff
@@ -0,0 +1,63 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+type byX struct {
+	x []float64
+	c []bool
+}
+
+func (a byX) Len() int { return len(a.x) }
+func (a byX) Swap(i, j int) {
+	a.x[i], a.x[j] = a.x[j], a.x[i]
+	a.c[i], a.c[j] = a.c[j], a.c[i]
+}
+func (a byX) Less(i, j int) bool { return a.x[i] < a.x[j] }
+
+// ROC sorts both inputs for increasing values in y, and
```

Comment:
Normally we start doc comments along the lines "ROC returns the ..." Would you prefix this?